lib/g5_authenticatable_api/services/token_info.rb

```diff
@@ -1,35 +1,51 @@
+# frozen_string_literal: true
+
 module G5AuthenticatableApi
   module Services
+    # Extract access token from request to retrieve token data from G5 Auth
     class TokenInfo
       attr_reader :params, :headers, :warden
 
-      def initialize(params={},headers={},warden=nil)
+      def initialize(params = {}, headers = {}, warden = nil)
         @params = params || {}
         @headers = headers || {}
         @warden = warden
       end
 
       def access_token
-        @access_token ||= (extract_token_from_header ||
-                           params['access_token'] ||
-                           warden.try(:user).try(:g5_access_token))
+        @access_token ||= begin
+                            extract_token_from_header ||
+                              extract_token_from_params ||
+                              extract_token_from_warden
+                          end
       end
 
       def token_data
         auth_client.token_info
       end
 
       def auth_client
-        @auth_client ||= G5AuthenticationClient::Client.new(allow_password_credentials: 'false',
-                                                            access_token: access_token)
+        @auth_client ||= G5AuthenticationClient::Client.new(
+          allow_password_credentials: 'false',
+          access_token: access_token
+        )
       end
 
       private
+
       def extract_token_from_header
-        if authorization_header
-          parts = authorization_header.match(/Bearer (?<access_token>\S+)/)
-          parts['access_token']
-        end
+        return unless authorization_header
```

Comment:
You've come over to the dark side, too. Early-returns are an anti-pattern in many other languages.

Reply:
Indeed. The first time rubocop suggested an early return with a guard clause, it felt so...wrong. I resisted for a long time. Even now, I think twice before I use it. But in those cases where you really do need some defensive programming (like when you're inspecting a request that could've come from anywhere), it makes the code much cleaner.